src/asset-management/AaveManager.sol

```diff
@@ -117,13 +117,15 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
         uint256 lShares = _decreaseShares(aPair, aToken, aAaveToken, aAmount);
         pool.withdraw(address(aToken), aAmount, address(this));
         emit FundsDivested(aPair, aToken, lShares);
-        aToken.safeApprove(address(aPair), aAmount);
+//        aToken.safeApprove(address(aPair), aAmount);
+        SafeTransferLib.safeApprove(address(aToken), address(aPair), aAmount);
```

Comment:
Are `solmate` and `solady` safeTransfer/safeApprove equivalent? E.g. perform all the same checks?

Reply:
`solmate`'s `safeTransfer`, `safeTransferFrom`, and `safeApprove`, checks if the `returndatasize` is greater than 31 bytes, `solady`'s doesn't. I guess this is to guard against implementations that return something but not exactly a `bool` or full 32 bytes? But they check that the return value is EXACTLY `1` so I think the check is not very necessary?

Both impl checks if the `returndatasize` is zero.

I'd imagine both implementations would have problems with `BNB`'s [implementation](https://etherscan.io/address/0xb8c77482e45f1f44de1745f52c74426c631bdd52#code) of `transfer` since it doesn't return anything?

Reply:
if it does not return anything then it is assumed to succeed, if it does reutrn something then you need to check if its a bool

Reply:
so are you saying that you agree with `solmate`'s additional check for a `returndatasize` greater than 31 bytes?  

Cuz the return value could be 1, and the size is greater than 0, but the type could be `int`, `uint`, `uint8` etc...

Reply:
Hmm, I'd want to know what use case solamte's extra check is covering. Though, I'm happy to use solady for now and have a task which states "answer why solmate checks data size". Which can be a low prio back log task

Reply:
tracked in #159, resolving for now